### PR TITLE
[chore] Incremental improvements to the release doc

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
More incremental improvements to the release doc.

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -72,6 +74,7 @@ The last step of the release process creates artifacts for the new version of th
...
 2. `commitChangesToNewBranch failed: invalid merge` -- This is a [known issue](https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/47) with our release tooling. The current workaround is to clone a fresh copy of the repository and try again. Note that you may need to set up a `fork` remote pointing to your own fork for the release tooling to work properly.
 3. `could not run Go Mod Tidy: go mod tidy failed` when running `multimod` -- This is a [known issue](https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/46) with our release tooling. The current workaround is to run `make gotidy` manually after the multimod tool fails and commit the result.
+4. If `make update-otel` produces errors in the CI [like these](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4812972510/jobs/8568980443?pr=21198#step:3:191), it could be because the make target was run too soon after updating Core and the goproxy hasn't update yet.  Try running `export GOPROXY=direct` and then `make update-otel`.
```
Comment (evan-bradley) [inline docs/release.md:77]: Will this log be kept forever? I think we should reproduce the relevant contents here if we can't make that guarantee.

File: docs/release.md [hunk 2]
```diff
@@ -27,25 +27,27 @@ It is possible that a core approver isn't a contrib approver. In that case, the
...
+5. Make sure you are on `release/<release-series>`. Tag all the module groups (stable, beta) with the new release version by running the `make push-tags` command (e.g. `make push-tags MODSET=stable` and `make push-tags MODSET=beta`). If you set your remote using `https` you need to include `REMOTE=https://github.com/open-telemetry/opentelemetry-collector.git` in each command. Wait for the new tag build to pass successfully.
 
 6. The release script for the collector builder should create a new GitHub release for the builder. This is a separate release from the core, but we might join them in the future if it makes sense.
 
 7. A new `v0.55.0` release should be automatically created on Github by now. Edit it and use the contents from the CHANGELOG.md as the release's description.
 
+8. If you created a draft PR to Contrib in step 3, updated the PR to use the newly released Core version and set it to Ready for Review. **Do not** move forward until this PR is merged.
```
Comment (evan-bradley) [inline docs/release.md:42]:
```suggestion
8. If you created a draft PR to Contrib in step 3, update the PR to use the newly released Core version and set it to Ready for Review. **Do not** move forward until this PR is merged.  
```

**Review by evan-bradley — COMMENTED** [review]
Thanks for going through and revising this.

File: docs/release.md (hunk 1 above)
Comment (TylerHelmuth) [reply docs/release.md:77]: Good question. I'll inline the error.

**Review by mx-psi — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**